WordPress/Sniffs/WP/DeprecatedParametersSniff.php

```diff
@@ -0,0 +1,347 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @link    https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+/**
+ * Check for usage of deprecated parameters in WP functions and sugests alternative based on the parameter passed.
+ *
+ * This sniff will throw an error when usage of deprecated parameters is
+ * detected if the parameter was deprecated before the minimum supported
+ * WP version; a warning otherwise.
+ * By default, it is set to presume that a project will support the current
+ * WP version and up to three releases before.
+ *
+ * @package WPCS\WordPressCodingStandards
+ *
+ * @since   0.11.0
+ */
+class WordPress_Sniffs_WP_DeprecatedParametersSniff extends WordPress_AbstractFunctionParameterSniff {
+
+	/**
+	 * The group name for this group of functions.
+	 *
+	 * @since 0.11.0
+	 *
+	 * @var string
+	 */
+	protected $group_name = 'wp_deprecated_parameters';
+
+	/**
+	 * Minimum WordPress version.
+	 *
+	 * This variable allows changing the minimum supported WP version used by
+	 * this sniff by setting a property in a custom ruleset xml file.
+	 *
+	 * Example usage:
+	 * <rule ref="WordPress.WP.DeprecatedParameters">
+	 *  <properties>
+	 *   <property name="minimum_supported_version" value="4.4"/>
+	 *  </properties>
+	 * </rule>
+	 *
+	 * @since 0.11.0
+	 *
+	 * @var string WordPress version.
+	 */
+	public $minimum_supported_version = 4.4;
+
+	/**
+	 * Array of function, argument, and replacement function for deprecated argument.
+	 *
+	 * @since 0.11.0
+	 *
+	 * @var array Multi-dimentional array with parameter details.
+	 *
+	 *               @type string Function name. {
+	 *                   @type int target Parameter positions. {
+	 *                       @type string|boolean|null  Default value.
+	 *                       @type int The WordPress version when deprecated.
+	 *                   }
+	 *               }
+	 */
+	protected $target_functions = array(
```

Comment:
I think all things considered, we should probably just have this list be in alphabetical order. I don't really see a compelling case for putting it in order of version number, so for simplicity and consistency alphabetical order seems best.